modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -253,10 +265,40 @@ struct IEUnit {
                 }
             }
         }
-#endif
-
+#endif // INF_ENGINE_RELEASE < 2020000000
+        // In dnn, after ordinary extensions they try to add custom <InfEngineExtension> as
+        // "some of networks can work without a library of extra layers.
+        //  OpenCV fallbacks as extensions"
+        // Do WE need this? (dnn/src/op_inf_engine.cpp:824)
+#if INF_ENGINE_RELEASE < 2019020000  // < 2019.R2
         auto this_network = this_plugin.LoadNetwork(net, {}); // FIXME: 2nd parameter to be
                                                               // configurable via the API
+#else
+        bool isHetero = false;
+        if (params.device_id != "CPU")
+        {
+            isHetero = params.device_id == "FPGA";
+            for (auto& layer : net)
+            {
+                if (layer->type == "OpenCVLayer")
+                {
+                    isHetero = true;
+#if INF_ENGINE_RELEASE < 2019030000  // < 2019.R3
+                    // Not sure about lower versions but in 2019R3 we do not need this
+                    layer->affinity = "CPU";
+                }
+                else
+                {
+                    layer->affinity = params.device_id;
+#endif
+                }
+            }
+        }
+        IE::ExecutableNetwork this_network = isHetero ?
+            core.LoadNetwork(net, "HETERO:" + params.device_id + ",CPU", {}) :
+            core.LoadNetwork(net, params.device_id, {});
```

Comment:
Took loading the network from dnn, can't figure if it's redundant

Reply:
`{}` can be omitted

Reply:
OK, omitted